smart_campus/app/views.py

```diff
@@ -1359,3 +1361,56 @@ def group_delete_page(request, pk):
     group_instance.delete()
 
     return HttpResponseRedirect('/groups/')
+
+
+@require_GET
+@csrf_exempt
+def get_unanswered_question(request):
+    station_id = request.GET.get('station_id')
+    email = request.GET.get('email')
+
+    station = Station.objects.filter(id=station_id).first()
+    user = User.objects.filter(email=email).first()
+
+    if not user or not station:
+        return HttpResponse('Either user or station does not exist', status=400)
+
+    unanswered_questions = Question.objects.exclude(user__pk=email).filter(
+        linked_station=station
+    )
+
+    random_index = random.randint(0, unanswered_questions.count() - 1)
+    random_unanswered_question = unanswered_questions[random_index]
```

Comment:
If my memory serves me right, you could do it using `random.sample`

Reply:
Yeah, but making a query with model in django should not privide a list type which random.sample will return.
I'm curious if there is a much better way.